src/utils/avatar.js

```diff
@@ -81,7 +81,6 @@ export class AvatarURL {
     } else {
       ensureUnreachable(rawAvatarUrl);
       const msg = 'Unexpected value for `rawAvatarUrl` in `AvatarURL.fromUserOrBotData`';
-      logging.error(msg, { value: rawAvatarUrl });
       throw new Error(msg);
```

Comment:
avatar: Remove redundant logging.error when we're already throwing an error
    
    This would produce log noise in the Jest tests in an upcoming commit.

Hmm, how do we end up reaching this code in the tests? The `ensureUnreachable` means the type-checker should verify it can't be reached given our types, and we generally write our tests to be well-typed.

I think the reason the `logging.error` is there in addition to the throw is that it adds the extra data of what the unexpected `rawAvatarUrl` was; that's what makes it not redundant with the throw. Happy to remove it if it's getting in the way somehow, though — it looks like we see zero cases of this error in Sentry.

Reply:
(Answered below: https://github.com/zulip/zulip-mobile/pull/5773#discussion_r1343172087)